### ISSUE TITLE
praisonai-train: remote stop signals one pid, so a stopped fine-tune keeps running on the GPU

### ISSUE THREAD

**Issue by MervinPraison** [MervinPraison/PraisonAI#4547]
`RemoteRunner.stop()` signals a single pid, so a stopped remote fine-tune leaves
its children running on the GPU.

The code's own comment says it writes the pid *"so `stop` can signal the whole
process group"* — and then `stop` never does.

**Where:** `src/praisonai-train/praisonai_train/remote/runner.py`, `stop()`

```python
f"  if kill -0 $pid 2>/dev/null; then "
f"    kill -TERM $pid 2>/dev/null; echo stopped; "
```

`kill -TERM $pid`, not `kill -TERM -$pgid`.

## Why that leaves work running

The pid recorded at launch is the wrapper shell, not the trainer. Reproduced
with the exact launch shape from `_launch_script`:

```
recorded pid: 17429
  17429  ppid 1      /bin/sh -c  cd … && nohup sh -c "…" &   ← what stop kills
  17430  ppid 17429  sh -c sleep 30 …                        ← the actual work
```

So `stop` kills 17429 and 17430 survives, reparented to init. A real fine-tune's
children are dataloader workers and torchrun ranks, all holding the card.

This is the same defect as #4491 (quitting orphaned the local fine-tune), which
was fixed for local runs. Remote runs still have it.

## Also: `stop` reports success for a kill that did not finish the job

`return result.stdout.strip() == "stopped"` is set by the shell as soon as
`kill -TERM` is *issued*. Nothing re-probes. A trainer that ignores SIGTERM, or
takes time to die, still reports `stopped` — so a UI can label a live paid run
"cancelled".

## Fix

Resolve the group on the host and signal it, then re-probe:

```sh
pid=$(cat "$PIDFILE")
g=$(ps -o pgid= -p $pid | tr -d ' ')
# never signal our own group
[ "$g" = "$(ps -o pgid= -p $$ | tr -d ' ')" ] && { echo refused; exit 1; }
kill -TERM -"$g"
sleep 2
kill -0 $pid 2>/dev/null && echo still-running || echo stopped
```

Return `True` only for `stopped`; raise on `still-running`.

Note `setsid` is **not** available on macOS, so the launcher cannot create a
group that way — resolving the real pgid on the host is the portable route.

## Test

`praisonai-train` has no remote tests that exercise a real process. A fake host
whose `run()` executes the script locally under `/bin/sh` is enough: launch a
script that spawns a child, call `stop`, and assert **the child is gone** — not
that `stop` returned True. Asserting the return value is what would have let
this through.
